test/ae_mdw_web/controllers/activities_controller_test.exs

```diff
@@ -324,6 +324,69 @@ defmodule AeMdwWeb.ActivitiesControllerTest do
       end
     end
 
+    test "when activities contain aexn tokens, it returns them as AexnEvent", %{conn: conn} do
+      account_pk = TS.address(0)
+      account = Enc.encode(:account_pubkey, account_pk)
+      another_account_pk = TS.address(1)
+      another_account = Enc.encode(:account_pubkey, another_account_pk)
+      height1 = 398
+      height2 = 399
+      txi1 = 123
+      txi2 = 456
+      txi3 = 789
+
+      store =
+        empty_store()
+        |> Store.put(
+          Model.AexnTransfer,
+          Model.aexn_transfer(index: {:aex9, account_pk, txi1, another_account_pk, 1, 1})
+        )
+        |> Store.put(Model.Tx, Model.tx(index: txi1, block_index: {height1, 0}, id: "hash1"))
+        |> Store.put(
+          Model.RevAexnTransfer,
+          Model.aexn_transfer(index: {:aex141, account_pk, txi2, another_account_pk, 2, 2})
+        )
+        |> Store.put(Model.Tx, Model.tx(index: txi2, block_index: {height2, 0}, id: "hash2"))
+        |> Store.put(
+          Model.RevAexnTransfer,
+          Model.aexn_transfer(index: {:aex9, account_pk, txi3, another_account_pk, 3, 3})
+        )
+        |> Store.put(Model.Tx, Model.tx(index: txi3, block_index: {height2, 0}, id: "hash3"))
+
+      assert %{"prev" => nil, "data" => [activity1, activity2], "next" => next_url} =
+               conn
+               |> with_store(store)
+               |> get("/v2/accounts/#{account}/activities", direction: "forward", limit: 2)
+               |> json_response(200)
+
+      assert %{
+               "height" => ^height1,
+               "type" => "Aex9TransferEvent",
+               "payload" => %{
+                 "from" => ^account,
+                 "to" => ^another_account,
+                 "value" => 1,
+                 "index" => 1
+               }
+             } = activity1
+
+      assert %{
+               "height" => ^height2,
+               "type" => "Aex141TransferEvent",
+               "payload" => %{
+                 "from" => ^another_account,
+                 "to" => ^account,
+                 "value" => 2,
+                 "index" => 2
+               }
+             } = activity2
+
+      assert %URI{query: query} = URI.parse(next_url)
+
+      assert %{"cursor" => _cursor, "direction" => "forward", "limit" => "2"} =
```

Comment:
would you mind adding a `get` to the `next_url` to validate the pagination cursor encoding and decoding?

Reply:
Done